cmd/namespace/delete.go

```diff
@@ -76,3 +90,71 @@ func (nc *NamespaceCommand) ExecuteDeleteNamespace(ctx context.Context, namespac
 	}
 	return nil
 }
+
+func (nc *NamespaceCommand) watchDelete(ctx context.Context, namespace string) error {
+	waitCtx, ctxCancel := context.WithCancel(ctx)
+	defer ctxCancel()
+
+	stop := make(chan os.Signal, 1)
+	defer close(stop)
+
+	signal.Notify(stop, os.Interrupt)
+	exit := make(chan error, 1)
+	defer close(exit)
+
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+	go func(wg *sync.WaitGroup) {
+		defer wg.Done()
+		exit <- nc.waitForNamespaceDeleted(waitCtx, namespace)
+	}(&wg)
+
+	wg.Add(1)
+	go func(wg *sync.WaitGroup) {
+		defer wg.Done()
+		err := nc.okClient.Stream().DestroyAllLogs(waitCtx, namespace)
+		if err != nil {
+			oktetoLog.Warning("delete namespace logs cannot be streamed due to connectivity issues")
+			oktetoLog.Infof("delete namespace logs cannot be streamed due to connectivity issues: %v", err)
+		}
+	}(&wg)
+
+	select {
+	case <-stop:
+		ctxCancel()
+		oktetoLog.Infof("CTRL+C received, exit")
+		return oktetoErrors.ErrIntSig
+	case err := <-exit:
+		// wait until streaming logs have finished
+		wg.Wait()
+		return err
+	}
+}
+
+func (nc *NamespaceCommand) waitForNamespaceDeleted(ctx context.Context, namespace string) error {
+	timeout := 5 * time.Minute
+	ticker := time.NewTicker(1 * time.Second)
+	to := time.NewTicker(timeout)
+
+	for {
+		select {
+		case <-to.C:
+			return fmt.Errorf("'%s' deploy didn't finish after %s", namespace, timeout.String())
+		case <-ticker.C:
+			ns, err := nc.k8sClient.CoreV1().Namespaces().Get(ctx, namespace, metav1.GetOptions{})
+			if err != nil {
+				// when err is NotFound return without error, as the namespace is correctly deleted
```

Comment:
here i have doubts. when the namespace is deleted i get the following error :

`namespaces "XXXXX" is forbidden: User "system:serviceaccount:okteto:XXXXXXXXX" cannot get resource "namespaces" in API group "" in the namespace "XXXX"`

i was expecting not found 🤷🏻‍♀️

Reply:
We can handle `k8serrors.IsNotFound` and `k8serrors.IsForbidden` as a success and include a comment indicating that the error returned by kubernetes is a forbidden in case the service account don't have access to all ns